x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/rule_details_flyout/tabs/translation/migration_rule_query.tsx

```diff
@@ -8,35 +8,36 @@
 import React, { useCallback, useMemo, useState } from 'react';
 import {
   EuiButtonEmpty,
+  EuiCodeBlock,
   EuiFlexGroup,
   EuiFlexItem,
   EuiHorizontalRule,
-  EuiMarkdownFormat,
   EuiSpacer,
   EuiTitle,
   useEuiTheme,
 } from '@elastic/eui';
 import { css } from '@emotion/react';
 import { VALIDATION_WARNING_CODES } from '../../../../../../detection_engine/rule_creation/constants/validation_warning_codes';
 import { useFormWithWarnings } from '../../../../../../common/hooks/use_form_with_warnings';
-import { EsqlQueryEdit } from '../../../../../../detection_engine/rule_creation/components/esql_query_edit';
 import { Field, Form, getUseField } from '../../../../../../shared_imports';
 import type { RuleTranslationSchema } from './types';
 import { schema } from './schema';
 import * as i18n from './translations';
+import { EsqlEditor } from './esql_editor';
 
 const CommonUseField = getUseField({ component: Field });
 
 interface MigrationRuleQueryProps {
   title: string;
   ruleName?: string;
   query: string;
+  isOriginalQuery?: boolean;
   canEdit?: boolean;
   onTranslationUpdate?: (ruleName: string, ruleQuery: string) => Promise<void>;
 }
 
 export const MigrationRuleQuery: React.FC<MigrationRuleQueryProps> = React.memo(
-  ({ title, ruleName, query, canEdit, onTranslationUpdate }) => {
+  ({ title, ruleName, query, canEdit, isOriginalQuery, onTranslationUpdate }) => {
```

Comment:
Consider passing the `language` directly as a prop instead. Original rules will come from different vendors in the future, we should not assume they will always be _spl_.

An easy approach: define this new `language` prop as an arbitrary _string_, them pass `original_rule.query_language` and `elastic_rule.query_language` values from the consumer. Inside this component we could map them to the best code block _language_:
 - `spl` -> `splunk-spl`
 - `esql` -> `sql`
 - more to come...
 - default -> undefined

Reply:
done in https://github.com/elastic/kibana/pull/206285/commits/a63cecc37daa67d9915b00df3ac8606769e6f1d0